README.md

```diff
@@ -17,30 +17,62 @@ void main() async {
+
 ```dart
 SpotifyApi getSpotifyApi() async {
   final credentials = SpotifyApiCredentials(clientId, clientSecret);
-  grant = SpotifyApi.authorizationCodeGrant(credentials);
-  // *** from dart-lang/oauth2 package
-  // `redirect` is an imaginary function that redirects the resource
-  // owner's browser.
-  await redirect(grant.getAuthorizationUrl(redirectUrl));
-  // Another imaginary function that listens for a request to `redirectUrl`.
-  final requestUri = await listen(redirectUrl);
-  // ***
-  return SpotifyApi.fromAuthCodeGrant(WelcomeScreen.grant, requestUri);
+  final grant = SpotifyApi.authorizationCodeGrant(credentials);
+
+  // The URI to redirect to after the user grants or denies permission.
+  // This URI must be in your Spotify application's Redirect URI whitelist.
+  final redirectUri = 'https://example.com/auth';
+
+  // See https://developer.spotify.com/documentation/general/guides/scopes/
+  // for a complete list of these Spotify authorization permissions. If no
+  // scopes are specified, only public Spotify information will be available.
+  final scopes = ['user-read-email', 'user-library-read'];
+
+  var authUri = grant.getAuthorizationUrl(
+    Uri.parse(redirectUri),
+    scopes: scopes, // scopes are optional
+  );
+
+  // `redirect` is an imaginary function that redirects the resource owner's
+  // browser to the `authUri` on the authorization server. Once the resource
+  // owner has authorized, they'll be redirected to the `redirectUri` with an
+  // authorization code. The exact implementation varies across platforms.
+  await redirect(authUri);
+
+  // `listen` is another imaginary function that listens for a request to
+  // `redirectUri` after the user grants or denies permission. Again, the
+  // exact implementation varies across platforms.
+  final responseUri = await listen(redirectUri);
+
+  return SpotifyApi.fromAuthCodeGrant(grant, responseUri);
 }
 ```
 
+Unfortunately, there's not a universal example for implementing the imaginary functions, `redirect` and `listen`, because different options exist for each platform.
+
+For a Flutter app, there's two popular approaches:
+1. Launch a browser using [url_launcher][url_launcher] and listen for a deeplink redirect using [uni_links][uni_links]. The device running the Flutter app will function as the authorization server and handle the redirect. The redirect URI should look something like `com.app://auth`.
+2. Launch a WebView inside the app and listen for a redirect using [webview_flutter][webview_flutter]. This approach requires redirecting to an external authorization server, such as a backend service or serverless function. The redirect URI should be a normal URL, like `https://example.com/auth`, that points to the authorization server.
+
```

Comment:
Nice! this clarifies the "imaginary functions" thing.
A third option using webview_flutter as well and listen to the redirect URL with `navigationDelegate`, like this the redirect URL can be made up and do not require backend service to actually listen to it.
```dart
WebView(
  javascriptMode: JavascriptMode.unrestricted,
  initialUrl: authUri,
  navigationDelegate: (navReq) {
    if (!navReq.url.startsWith(redirectUrl))
      return NavigationDecision.navigate;
    final api = SpotifyApi.fromAuthCodeGrant(grant, navReq.url);
    // do somthing with api
    return NavigationDecision.prevent;
  },  ...
```

Reply:
That's awesome, I'll add that approach to the list!